dashboard/config/initializers/statsig.rb

```diff
@@ -0,0 +1,5 @@
+require 'statsig'
+
+local_mode = CDO.rack_env?(:production) || CDO.rack_env?(:staging) ? false : true
+options = StatsigOptions.new({'tier' => CDO.rack_env}, network_timeout: 5, local_mode: local_mode)
+Statsig.initialize(CDO.statsig_server_secret_key, options)
```

Comment:
In our managed environments (adhocs, staging, test, levelbuilder, production), we use the `puma` Rack web application server to process multiple HTTP requests simultaneously across multiple child puma Unix processes, and multiple threads per child puma process. It looks like [there are special initialization](https://docs.statsig.com/server/rubySDK) instructions for StatSig when using `puma` to ensure it is initialized in each child `puma` process when the child process boots up:

``` ruby
on_worker_boot do
  Statsig.initialize('server-secret-key', options)
end
```

Here’s where we implement worker boot initialization logic: https://github.com/code-dot-org/code-dot-org/blob/92a89e8d1d94cdd6eda58543965e26e2f250c467/dashboard/config/puma.rb#L29-L32

And if we want StatSig on Pegasus as well, we could put that in this shared module https://github.com/code-dot-org/code-dot-org/blob/92a89e8d1d94cdd6eda58543965e26e2f250c467/lib/cdo/app_server_hooks.rb#L29-L32

Reply:
Thanks @sureshc, I'll move the initialization over to Puma. Should I go ahead and initialize it for Pegasus while I'm at it, that way it's available for anyone to use?

Reply:
Thinking through this, it seems like we should initialize in puma.rb, and then conditionally as an initializer if it's not a managed environment?

Statsig is [idempotent in it's initialization](https://github.com/statsig-io/ruby-sdk/blob/main/lib/statsig.rb#L14) (will only initialize if not already initialized). I'm not sure of the order of Rails bootstrapping. Does puma get start _before_ the initializers? Based on the [Rails guide](https://guides.rubyonrails.org/v6.1/initialization.html), it seems initializers get called first, so we would need to start Statsig in the initializers _only_ for non-managed environments, otherwise it wouldn't ever get initialized in Puma, since it would already be running.